### ISSUE TITLE
dropbox: getting "Couldn't move: move failed: from_lookup/not_found/" errors with --backup-dir and changed modtime

### ISSUE THREAD

**Issue by ncw** [ncw/rclone#2134]
#### What is the problem you are having with rclone?

See [the forum](https://forum.rclone.org/t/error-file-name-couldnt-move/5019) for background.

When using dropbox with `--backup-dir` if a file has only changed modtime, then in the first pass rclone deletes it then returns the error "Couldn't move: move failed: from_lookup/not_found/".

#### What is your rclone version (eg output from `rclone -V`)

```
rclone v1.39-233-g383feca6
- os/arch: linux/amd64
- go version: go1.10
```

#### Which OS you are using and how many bits (eg Windows 7, 64 bit)

Linux 64 Bit

####  Which cloud storage system are you using? (eg Google Drive)

Dropbox

#### The command you were trying to run (eg `rclone copy /tmp remote:tmp`)

    rclone -vv copy file.txt --backup-dir dropbox:test/test2 dropbox:test/test1

#### A log from the command with the `-vv` flag (eg output from `rclone -vv copy /tmp remote:tmp`)

Here is a reproduction of the problem with log

```
$ echo "hello" > file.txt
$ rclone -v copy file.txt dropbox:test/test1
2018/03/13 09:37:30 INFO  : Dropbox root 'test/test1': Modify window is 1s
2018/03/13 09:37:30 INFO  : Dropbox root 'test/test1': Waiting for checks to finish
2018/03/13 09:37:30 INFO  : Dropbox root 'test/test1': Waiting for transfers to finish
2018/03/13 09:37:31 INFO  : file.txt: Copied (new)
2018/03/13 09:37:31 INFO  : 
Transferred:      6 Bytes (3 Bytes/s)
Errors:                 0
Checks:                 0
Transferred:            1
Elapsed time:        1.9s

$ touch file.txt 

$ rclone -vv copy file.txt --backup-dir dropbox:test/test2 dropbox:test/test1 
2018/03/13 09:38:05 DEBUG : Using config file from "/home/ncw/.rclone.conf"
2018/03/13 09:38:05 DEBUG : rclone: Version "v1.39-233-g383feca6" starting with parameters ["rclone" "-vv" "copy" "file.txt" "--backup-dir" "dropbox:test/test2" "dropbox:test/test1"]
2018/03/13 09:38:06 INFO  : Dropbox root 'test/test1': Modify window is 1s
2018/03/13 09:38:07 INFO  : Dropbox root 'test/test1': Waiting for checks to finish
2018/03/13 09:38:07 DEBUG : file.txt: Modification times differ by -45.652019585s: 2018-03-13 09:37:38.652019585 +0000 GMT, 2018-03-13 09:36:53 +0000 UTC
2018/03/13 09:38:07 DEBUG : file.txt: src and dst identical but can't set mod time without deleting and re-uploading
2018/03/13 09:38:08 ERROR : file.txt: Couldn't move: move failed: from_lookup/not_found/...
2018/03/13 09:38:08 INFO  : Dropbox root 'test/test1': Waiting for transfers to finish
2018/03/13 09:38:08 ERROR : Attempt 1/3 failed with 1 errors and: move failed: from_lookup/not_found/...
2018/03/13 09:38:10 INFO  : Dropbox root 'test/test1': Waiting for checks to finish
2018/03/13 09:38:10 INFO  : Dropbox root 'test/test1': Waiting for transfers to finish
2018/03/13 09:38:10 INFO  : file.txt: Copied (new)
2018/03/13 09:38:10 ERROR : Attempt 2/3 succeeded
2018/03/13 09:38:10 INFO  : 
Transferred:      6 Bytes (1 Bytes/s)
Errors:                 0
Checks:                 1
Transferred:            1
Elapsed time:          5s

2018/03/13 09:38:10 DEBUG : 7 go routines active
2018/03/13 09:38:10 DEBUG : rclone: Version "v1.39-233-g383feca6" finishing with parameters ["rclone" "-vv" "copy" "file.txt" "--backup-dir" "dropbox:test/test2" "dropbox:test/test1"]

$ rclone lsl dropbox:test/test1 # file is correct here
        6 2018-03-13 09:37:39.000000000 file.txt

$ rclone lsl dropbox:test/test2 # nothing in the backup
2018/03/13 09:38:58 ERROR : : error listing: directory not found
2018/03/13 09:38:58 Failed to lsl: directory not found

$ ls -l --full-time file.txt
-rw-rw-r-- 1 ncw ncw 6 2018-03-13 09:37:38.652019585 +0000 file.txt
```

#### Solution

I think in an ideal world we'd like rclone to put the incorrectly dated `file.txt` into the `--backup-dir`, but at minimum we want rclone to transfer the file straight away and not wait for the retry.

**Comment by TowerBR** [comment]
Perhaps the solution is to put an ```if``` in the tests:

> this file has incorrect modification time, so "I" will have to re-upload
> 
>>  is -backup-dir set?
>>>  yes!
>>>  then move this incorrectly dated file into the --backup-dir
>>>  no!
>>>  delete the file
>
> upload the file again

**Comment by ncw** [comment]
I've fixed this in a way not too disimilar to your suggestion @TowerBR :-)

It worked for my tests - please re-open if it doesn't work for you.

https://beta.rclone.org/v1.39-234-gdfd0f4c5/ (uploaded in 15-30 mins)

**Comment by TowerBR** [comment]
Thank you for the quick solution!

**Comment by TowerBR** [comment]
@ncw , this bug fix in v1.40:

> sync: when using --backup-dir don’t delete files if we can’t set their modtime
>         this fixes odd behaviour with Dropbox and --backup-dir

Is it the same subject that we were discussing here in this topic?

If yes, if the files are not deleted, what exactly happens?

**Comment by ncw** [comment]
@TowerBR yes that is this bug :-)
